registrations/views.py

```diff
@@ -166,14 +166,22 @@ def post(self, request):
             api_url=settings.IDENTITY_STORE_URL,
             auth_token=settings.IDENTITY_STORE_TOKEN
         )
-        # load the users sources - posting users should only have one source
-        source = Source.objects.get(user=self.request.user)
         serializer = ThirdPartyRegistrationSerializer(data=request.data)
         if serializer.is_valid():
             mom_msisdn = serializer.validated_data['mom_msisdn']
             hcw_msisdn = serializer.validated_data['hcw_msisdn']
             lang_code = serializer.validated_data['mom_lang']
             lang_code = transform_language_code(lang_code)
+            authority = serializer.validated_data['authority']
+            # load the users sources with authority mapping
+            if authority == 'chw':
+                source_auth = 'hw_limited'
+            elif authority == 'clinic':
+                source_auth = 'hw_full'
+            else:
+                source_auth = 'patient'
+            source = Source.objects.get(user=self.request.user,
+                                        authority=source_auth)
```

Comment:
@gsvr This lookup will support a single user with multiple sources. Are there any implications on other parts of the code though? I think as long as the source and user used for external registrations is never used elsewhere is should be fine right?